### PR TITLE
Read network device names from Network Manager

### PR DESCRIPTION
**Pull request by ilmanzo**
Fix for openSUSE, makes the test independent from the network device naming scheme

- Related ticket: https://progress.opensuse.org/issues/151273
- Needles: N/A
- Verification runs: 
  - https://openqa.opensuse.org/tests/3800277
  - https://openqa.suse.de/tests/13023338

### CODE REVIEW

**Comment by michaelgrifalconi** [comment]
Is there any other way instead of grep-ing text ouputs meant for user interface? Like some json output for example, to be more on the safe side (I am not a fan of using grep output as an API as might change, even if unlikely). 
But on the other hand is an easy thing to fix if it will ever break, so no hard opinions here!

File: tests/microos/network_bonding.pm [hunk 1]
```diff
@@ -27,11 +27,14 @@ sub run {
     my ($self) = @_;
     select_console 'root-console';
     # remove existing NM-managed connections (except loopback)
-    my @interfaces = grep { !/^lo/ } split('\n', script_output 'nmcli -t -f NAME con');
-    assert_script_run "nmcli con delete '$_'" for @interfaces;
+    my @connections_info = grep { !/^lo/ } split('\n', script_output 'nmcli -t -f DEVICE,UUID conn show --active');
```
Comment (Vogtinator) [inline tests/microos/network_bonding.pm:30]: FWICT you can use

```suggestion
    my @connections_info = grep { !/^lo/ } split('\n', script_output 'nmcli -g DEVICE,UUID conn show --active');
```

File: tests/microos/network_bonding.pm [hunk 2]
```diff
@@ -27,11 +27,14 @@ sub run {
...
+    my @devices = map { (split /:/)[0] } @connections_info;
+    my @connection_ids = map { (split /:/)[1] } @connections_info;
+    # can fail due to same ID pointing to different connections
+    script_run "nmcli con delete '$_'" for @connection_ids;
```
Comment (Vogtinator) [inline tests/microos/network_bonding.pm:34]: IMO ignoring failures should be avoided if possible. Can you uniq `@connection_ids`?

**Comment by ilmanzo** [comment]
> Is there any other way instead of grep-ing text ouputs meant for user interface? Like some json output for example, to be more on the safe side (I am not a fan of using grep output as an API as might change, even if unlikely). But on the other hand is an easy thing to fix if it will ever break, so no hard opinions here!

I'm not aware of any json output (but it can be a good idea to add in `nmcli` :) according to the man page, 
```
 -t | --terse
           Output is terse. This mode is designed and suitable for computer (script) processing.
```
we can also use `-g` as @Vogtinator suggests, but it's pretty much an alias

File: tests/microos/network_bonding.pm (hunk 2 above)
Comment (ilmanzo) [reply tests/microos/network_bonding.pm:34]: good point ! Thanks

**Comment by michaelgrifalconi** [comment]
Oh interesting, so they are aware that this is used in scripts and provide some sort of API-like text output :D Good enough I guess :)

**Review by ricardobranco777 — APPROVED** [review]

**Pull request merged**